src/solver/optimisation/constraint_builder.h

```diff
@@ -0,0 +1,376 @@
+    }
+
+    int IntercoDirectCost(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariableCoutOrigineVersExtremiteDeLInterconnexion[index];
+    }
+
+    int IntercoIndirectCost(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariableCoutExtremiteVersOrigineDeLInterconnexion[index];
+    }
+
+    int ShortTermStorageInjection(unsigned int index) const
+    {
+        return nativeOptimVar.SIM_ShortTermStorage.InjectionVariable[index];
+    }
+
+    int ShortTermStorageWithdrawal(unsigned int index) const
+    {
+        return nativeOptimVar.SIM_ShortTermStorage.WithdrawalVariable[index];
+    }
+
+    int ShortTermStorageLevel(unsigned int index) const
+    {
+        return nativeOptimVar.SIM_ShortTermStorage.LevelVariable[index];
+    }
+
+    int HydProd(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariablesDeLaProdHyd[index];
+    }
+
+    int HydProdDown(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariablesVariationHydALaBaisse[index];
+    }
+
+    int HydProdUp(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariablesVariationHydALaHausse[index];
+    }
+
+    int Pumping(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariablesDePompage[index];
+    }
+
+    int HydroLevel(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariablesDeNiveau[index];
+    }
+
+    int Overflow(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariablesDeDebordement[index];
+    }
+
+    int FinalStorage(unsigned int index) const
+    {
+        return NumeroDeVariableStockFinal[index];
+    }
+
+    int LayerStorage(unsigned area, unsigned layer) const
+    {
+        return NumeroDeVariableDeTrancheDeStock[area][layer];
+    }
+
+    int PositiveUnsuppliedEnergy(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariableDefaillancePositive[index];
+    }
+
+    int NegativeUnsuppliedEnergy(unsigned int index) const
+    {
+        return nativeOptimVar.NumeroDeVariableDefaillanceNegative[index];
+    }
+
+private:
+    const CORRESPONDANCES_DES_VARIABLES& nativeOptimVar;
+    const std::vector<int>& NumeroDeVariableStockFinal;
+    const std::vector<std::vector<int>>& NumeroDeVariableDeTrancheDeStock;
+};
+
+} // namespace Variable
+
+class ConstraintBuilder
+{
+public:
+    ConstraintBuilder(
+      PROBLEME_HEBDO& problemeHebdo,
+      const std::vector<CORRESPONDANCES_DES_VARIABLES>& CorrespondanceVarNativesVarOptim) :
+     problemeHebdo(problemeHebdo),
+     problemeAResoudre(*problemeHebdo.ProblemeAResoudre),
+     varNative(CorrespondanceVarNativesVarOptim)
+    {
+    }
+
+    ConstraintBuilder& updateHourWithinWeek(unsigned hour)
+    {
+        hourInWeek_ = hour;
+        return *this;
+    }
+
+public:
+    ConstraintBuilder& DispatchableProduction(unsigned int index,
```

Comment:
I think we should add some documentation, either at the method or class level, to explain the reader what's the role of the class/method, and in particular what are the non obvious things:
- what is the difference between shift and delta ?
- what means wrapping ?
- ...

(although already the refactoring documents the code better than before :) )